indico/modules/events/registration/fields/choices_test.py

```diff
@@ -0,0 +1,145 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from copy import deepcopy
+
+import pytest
+from indico.modules.events.registration.models.form_fields import RegistrationFormField
+from indico.modules.events.registration.models.registrations import RegistrationData
+
+
+def _id(n):
+    assert 0 <= n < 10
+    return '{}0000000-0000-0000-0000-000000000000'.format(n)
+
+
+@pytest.fixture
+def multi_choice_field():
+    field = RegistrationFormField(input_type='multi_choice')
+    field.versioned_data = {
+        'choices': [
+            {'id': _id(1), 'places_limit': 0, 'is_billable': False, 'price': 0},
+            {'id': _id(2), 'places_limit': 0, 'is_billable': False, 'price': 0},
+            {'id': _id(3), 'places_limit': 0, 'is_billable': True, 'price': 10}
+        ]
+    }
+    return field
+
+
+def _update_data(data, changes):
+    data = dict(deepcopy(data))
```

Comment:
Why are you using `dict` here? Looking at the code I think you are always passing a dict so `deepcopy` will return an object of the same type.